specs/market-api.yaml

```
@@ -827,27 +967,33 @@ components:
 
```

To enable querying only for events regarding a single agreement

```suggestion
    queryAgreementId:
      name: agreementId
      description: Used to enable querying agreement events related to a single agreement.
      in: query
      required: false
      schema:
        type: string

```

---

For sure it is worth considering, because we have no easy way to query, for example, termination reason for specific Agreement, without querying all events, even these unrelated.
But note that adding new filter to endpoint is backward compatible change, so I suggest to postpone these changes for later and investigate other options.

---

postponed to #53